core/src/test/java/io/questdb/mp/ConcurrentTest.java

```diff
@@ -77,6 +78,127 @@ public void testConcurrentFanOutAnd() {
         Assert.assertEquals(threads * iterations, doneCount.get());
     }
 
+    @Test
+    public void testConcurrentFanOutAsyncOffloadPattern() {
+        // This test aims to reproduce a simplified variant of async offload inter-thread communication pattern.
+
+        class Item {
+            int owner;
+        }
+
+        final int capacity = 2;
+        final int collectorThreads = 2;
+        final int reducerThreads = 2;
+        final int itemsToDispatch = 8;
+        final int iterations = 100;
+
+        final RingQueue<Item> queue = new RingQueue<Item>(() -> new Item(), capacity);
```

Comment:
nit: there is intellij warning on this line, lambda can be simplified

Reply:
Yes, but this breaks JDK 8 build.

Reply:
indeed, it was an issue with my IDE config